webui/react/src/e2e/models/hew/DataGrid.ts

```diff
@@ -0,0 +1,330 @@
+    return (index + 1) * this.parentTable.columnHeight + 5;
+  }
+
+  /**
+   * Clicks an x coordinate on the row
+   * @param {object} xPos - coordinate
+   */
+  async clickX(xPos: number): Promise<void> {
+    await this.parentTable.pwLocator.click({
+      position: { x: xPos, y: this.getY(await this.getIndex()) },
+    });
+  }
+
+  /**
+   * Clicks the row's select button
+   */
+  async clickSelect(): Promise<void> {
+    await this.clickX(5);
+  }
+
+  /**
+   * Returns a cell from an index. Start counting at 0.
+   */
+  getCellByIndex(n: number): BaseComponent {
+    return new BaseComponent({
+      parent: this,
+      selector: `[aria-colindex="${n + 1}"]`,
+    });
+  }
+
+  /**
+   * Returns a cell from a column name.
+   */
+  getCellByColumnName(s: string): BaseComponent {
+    const map = this.parentTable.headRow.columnDefs;
+    const index = map.get(s);
+    if (index === undefined) {
+      throw new Error(`Column with title expected but not found ${map}`);
+    }
+    return this.getCellByIndex(index - 1);
+  }
+}
+
+/**
+ * Returns the representation of a Table HeadRow.
+ * This constructor represents the Table in hew/src/kit/DataGrid.tsx.
+ * @param {object} obj
+ * @param {CanBeParent} obj.parent - The parent used to locate this HeadRow
+ * @param {string} obj.selector - Used as a selector uesd to locate this object
+ */
+export class HeadRow extends NamedComponent {
+  readonly defaultSelector = 'tr';
+  readonly clickableParentLocator: Locator;
+  constructor(args: HeadRowArgs) {
+    super(args);
+    this.clickableParentLocator = args.clickableParentLocator;
+  }
+
+  readonly selectDropdown: HeaderDropdown = new HeaderDropdown({
+    parent: this,
+    selector: '[aria-colindex="1"]',
+  });
+  #columnDefs = new Map<string, number>();
+
+  get columnDefs(): Map<string, number> {
+    if (this.#columnDefs.size === 0) {
+      throw new Error('Please set the column definitions using setColumnDefs first!');
+    }
+    return this.#columnDefs;
+  }
+
+  /**
+   * Sets Column Definitions
+   * Row.getCellByColumnName will fail without running this first.
+   */
+  async setColumnDefs(): Promise<Map<string, number>> {
+    const cells = await this.pwLocator.locator('th').all();
+    if (cells.length === 0) {
+      throw new Error('Expected to see more than 0 columns.');
+    }
+    await Promise.all(
+      cells.map(async (cell) => {
+        try {
+          const index = await cell.getAttribute('aria-colindex');
+          if (index === null) throw new Error();
+          this.#columnDefs.set(await cell.innerText(), +index);
+        } catch {
+          Promise.reject(
+            new Error(`All header cells should have the attribute ${'aria-colindex'}`),
+          );
+        }
+      }),
+    );
+    return this.#columnDefs;
+  }
+
+  /**
+   * Clicks the head row's select button
+   */
+  async clickSelectDropdown(): Promise<void> {
+    await this.clickableParentLocator.click({ position: { x: 5, y: 5 } });
```

Comment:
Is this OK on smaller screens? I know we haven't done much with it, but it'd be good to think about.

Reply:
ehh let's deal with that when we get to it